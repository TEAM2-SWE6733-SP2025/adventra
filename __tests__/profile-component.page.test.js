import { render, screen } from "@testing-library/react";
import ProfilePage from "../src/app/profile/page";
import "@testing-library/jest-dom";

jest.mock("../src/app/components/Navbar", () => {
  const Navbar = () => <div data-testid="navbar">Navbar</div>;
  Navbar.displayName = "Navbar"; // Set display name
  return Navbar;
});

jest.mock("../src/app/components/Profile", () => {
  const Profile = () => <div data-testid="profile">Profile</div>;
  Profile.displayName = "Profile"; // Set display name
  return Profile;
});

describe("ProfilePage", () => {
  it("renders the Navbar and Profile components", () => {
    render(<ProfilePage />);

    // Check if Navbar is rendered
    const navbar = screen.getByTestId("navbar");
    expect(navbar).toBeInTheDocument();

    // Check if Profile is rendered
    const profile = screen.getByTestId("profile");
    expect(profile).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<ProfilePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
