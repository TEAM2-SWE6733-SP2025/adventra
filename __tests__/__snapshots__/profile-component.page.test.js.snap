// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="block w-full"
  >
    <div
      data-testid="navbar"
    >
      Navbar
    </div>
  </header>
  <main
    class="flex justify-center items-center min-h-screen px-4 dark"
  >
    <div
      data-testid="profile"
    >
      Profile
    </div>
  </main>
</DocumentFragment>
`;
